pkg/kubectl/cmd/delete.go

```diff
@@ -248,13 +261,20 @@ func ReapResult(r *resource.Result, f cmdutil.Factory, out io.Writer, isDefaultD
 		if err != nil {
 			return err
 		}
+		// Set WaitForDeletion to false for namespace deletion unless it was explicitly set by the user.
+		if info.Mapping.Resource == "Namespace" && isDefaultWaitForDeletion {
```

Comment:
Users dont know whether cascading deletion is client-side using Reapers or server-side using finalizers. For them, `kubectl delete` returning means that the resource is deleted except for namespaces and pod graceful deletion.

Ref https://github.com/kubernetes/kubernetes/pull/40714#issuecomment-278460041 for where this was discussed before.